Update the cheetsheet (#889)
netdata/learn: docs/cheatsheet.mdx

description: "Connecting an Agent to the Cloud allows a Netdata Agent, running on a distributed node, to securely connect to Netdata Cloud via the encrypted Agent-Cloud link (ACLK)."
image: /cheatsheet/cheatsheet-meta.png
sidebar_label: Cheatsheet
custom_edit_url: https://github.com/netdata/learn/blob/master/docs/cheatsheet.mdx
---

Use our management &amp; configuration cheatsheet to simplify your interactions with Netdata, including configuration,
using charts, managing the daemon, and more.

> <div style={{ padding: '2px', float: 'left', width: '48%'}}>

Reviewer: I know that we thought of this two column thing so that it still looks like a traditional cheat sheet. However, I assume the reason why a cheatsheet is multi column, is that everything fits on **one** page. 
Currently, this is not the case. If you use ctrl+P, you end up with 6 pages, and because of the two column layout, the commands are often cut off. 

I would scrap the two columns altogether to be honest - we decided to go digital for a few reasons, so we shouldn't try to mime a print layout. 😄 

What is your opinion?

Author: Sure, along with the other points you made, I get that we expect the user to be online while he accesses the cheatsheet, so it makes sense to have it in one column, I used the two column layout to replicate the old design, but sure it makes more sense to have one column, cleaner code too

## Install Netdata

#### Install Netdata

```
wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh
```

Or, if you have cURL but not wget (such as on macOS):

```
curl https://my-netdata.io/kickstart.sh > /tmp/netdata-kickstart.sh && sh /tmp/netdata-kickstart.sh
```

#### Claim a node to Netdata Cloud

```
wget -O /tmp/netdata-kickstart.sh https://my-netdata.io/kickstart.sh && sh /tmp/netdata-kickstart.sh --claim-token TOKEN --claim-rooms ROOM1,ROOM2 --claim-url https://app.netdata.cloud
```

> > Sign in to Netdata Cloud, click the `Claim Nodes` button, choose the `War Rooms` to add nodes to, then click `Copy` to copy the full script to your clipboard. Paste that into your node’s terminal and run it.

Author: I agree on this, given we don't provide troubleshooting for offline scenarios in the cheatsheet, it makes sense to cut the info we give to save space 🙌 and explain a bit what every command does

> 

Reviewer: After the first logical step (Download & Claim) I would continue with "How to access the Netdata dashboard" (currently in line 150) and then "Manage the daemon".

Author: Ah yes, ordering has been flipped because we wanted the tables on the right (if you recall our talk with David), +1 on this

## Metrics collection & retention

#### Increase metrics retention (4GiB)

```

> sudo ./edit-config netdata.conf

Author: sure, if we suppose that the user doesn't print the cheatsheet (so we don't expect him to see it also offline) if he does, then some info will be unreachable. Saying this because the old one didn't have many links, but I do agree with what you are saying, just raising a point also

```

```
[global]
 dbengine multihost disk space = 4096
```

#### Reduce the collection frequency (every 5 seconds)

```
sudo ./edit-config netdata.conf
```

```
[global]
 update every = 5
```

#### Enable/disable plugins (groups of collectors)

```
sudo ./edit-config netdata.conf
```

```
[plugins]
 go.d = yes # enabled
 node.d = no # disabled
```

#### Enable/disable specific collectors

```
sudo ./edit-config go.d.conf
```

> `Or python.d.conf, node.d.conf, edbpf.conf, and so on`.

```
modules:
 activemq: no # disabled
 bind: no # disabled
 cockroachdb: yes # enabled
 ```

> #### Edit a collector s config (example)

Reviewer: ```suggestion
#### Edit a collector's config (example)
```

```
$ sudo ./edit-config go.d/mysql.conf
$ sudo ./edit-config ebpf.conf
$ sudo ./edit-config python.d/anomalies.conf
```

> ## Alarms & notifications

Reviewer: I think sometime ago we agreed to use the term "alert" instead of alarm, so it would be good if you could replace all instances. :)

Author: OH a stray one!!

#### Add a new alarm

```

> sudo touch health.d/example-alarm.conf

Reviewer: Do we assume here that the user is in the correct directory? Do we need to give additional context or does the explanation in "Configuration" suffice?

Author: We should put the "Configuration" part before this, does it make sense? Then we shouldn't need to explain anything

Reviewer: Yep makes sense. :)

> sudo ./edit-config health.d/example-alarm.conf

Reviewer: To be exact, only the first line creates the alarm file, correct? 

So I wouldn't include the second line because you mention it right beneath anyways. :)

```

#### Configure a specific alarm

```
sudo ./edit-config health.d/example-alarm.conf
```

#### Silence a specific alarm

```
sudo ./edit-config health.d/example-alarm.conf
 to: silent
```

#### Disable alarms and notifications

```
[health]
 enabled = no
```

> After any change, reload the Netdata health configuration

```
netdatacli reload-health
```

or if that command doesn't work on your installation, use:

```
killall -USR2 netdata
```
</div>

<div style={{ padding: '2px', float: 'right', width: '48%'}}>

## Manage the daemon

> | Intent                      |                                                                Action |

Reviewer: Very nice table. Reviewer is happy. 🐶

Author: 🙌🙌

|:----------------------------|----------------------------------------------------------------------:|
| Start Netdata               |                                      `$ sudo systemctl start netdata` |
| Stop Netdata                |                                       `$ sudo systemctl stop netdata` |
| Restart Netdata             |                                    `$ sudo systemctl restart netdata` |
| Reload health configuration | `$ sudo netdatacli reload-health` <br></br> `$ killall -USR2 netdata` |
| View error logs             |                                     `less /var/log/netdata/error.log` |

## See metrics and dashboards

#### Netdata Cloud `https://app.netdata.cloud`

#### Local dashboard `https://NODE:19999`

> Replace `NODE` with the IP address or hostname of your node. Often `localhost`.

#### Access the Netdata API `http://NODE:19999/api/v1/info`

## Configuration

> #### The Netdata config directory: `/etc/netdata`

Reviewer: I like the separation of the explanation from the filepath with a `:`

Should we do it in all headings with code?

Author: 👍

> > View the running Netdata configuration if your config directory is not here. The cheatsheet assumes you’re running all commands from within the Netdata config directory!

Reviewer: ```suggestion
> If you don't have such a directory:
> 📄 [Find your netdata.conf file](https://learn.netdata.cloud/guides/step-by-step/step-04#find-your-netdataconf-file)
> The cheatsheet assumes you’re running all commands from within the Netdata config directory!
```

Added a suggestion to make it more actionable.

#### Edit Netdata's main config file `$ sudo ./edit-config netdata.conf`

#### Edit Netdata's other config files (examples):

- `$ sudo ./edit-config apps_groups.conf`
- `$ sudo ./edit-config ebpf.conf`
- `$ sudo ./edit-config health.d/load.conf`
- `$ sudo ./edit-config go.d/prometheus.conf`

> #### View the running Netdata configuration `http://NODE:19999/netdata.conf`

Reviewer: We could move it up if you prefer to stick with the original phrasing of your note. (line 165)
